pkg/writer/serializer_cdx14.go

```diff
@@ -182,43 +200,80 @@ func nodeToCDX14Component(n *sbom.Node) *cdx14.Component {
 	}
 
 	if n.Licenses != nil && len(n.Licenses) > 0 {
-		c.Licenses = []cdx14.License{}
+		var licenseChoices []cdx14.LicenseChoice
+		var licenses cdx14.Licenses
 		for _, l := range n.Licenses {
-			c.Licenses = append(c.Licenses, cdx14.License{
-				License: struct {
-					ID string "json:\"id\"" // TODO optimize
-				}{l},
+			licenseChoices = append(licenseChoices, cdx14.LicenseChoice{
+				License: &cdx14.License{
+					ID: l,
+				},
 			})
 		}
+
+		licenses = licenseChoices
+		c.Licenses = &licenses
 	}
 
 	if n.Hashes != nil && len(n.Hashes) > 0 {
-		c.Hashes = []cdx14.Hash{}
+		c.Hashes = &[]cdx14.Hash{}
 		for algo, hash := range n.Hashes {
-			c.Hashes = append(c.Hashes, cdx14.Hash{
-				Algorithm: algo, // Fix to make it valid
-				Content:   hash,
+			*c.Hashes = append(*c.Hashes, cdx14.Hash{
+				Algorithm: NormalizeAlgo(algo), // Fix to make it valid
```

Comment:
Awesome, another one you knocked off my todo-list :+1: 

I think we should abstract this away to share the algo normalization function betwe reader+writers plus other CDX versions?